python_scripts/ensemble.py

```diff
@@ -923,15 +933,16 @@ def plot_decision_function(X, y, clf, sample_weight=None, fit=True, ax=None):
 #
 # ### Gradient-boosting decision tree
 #
-# For gradient-boosting, parameter tuning is a combination of several
-# parameters instead of setting one after the other each parameter. The
+# For gradient-boosting, parameters are coupled, so we can not anymore
+# set the parameters one after the other. The
 # important parameters are `n_estimators`, `max_depth`, and `learning_rate`.
 #
 # Let's first discuss the `max_depth` parameter. We saw in the section on
 # gradient-boosting that the algorithm fits the error of the previous tree
 # in the ensemble. Thus, fitting fully grown trees will be detrimental. Indeed,
 # the first tree of the ensemble would perfectly fit (overfit) the data and
 # thus no subsequent tree would be required, since there would be no residuals.
+# Instead, having very weak learners at each step will help reducing overfitting.
```

Comment:
Hmm doesn't seem to fit here as we talk above about residuals. Maybe move this to the end of the paragraph.